code/script.js

```diff
@@ -223,49 +229,67 @@ const generateBoard = () => {
 }
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
+// return Math.floor(Math.random() * (max - min + 1) + min)
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  secret = charactersInPlay[Math.floor(Math.random() * (charactersInPlay.length -1 - 0 + 1) + 0)];
+  console.log(secret);
 }
 
 // This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
+  /* Here we're setting charactersInPlay array to be all the characters to start with, we are also generating the board, and setting the secret
+  character for the game, the win/lose wrapper is not displayed until the user guessed. */
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  winOrLoseWrapper.style.display = 'none';
+  generateBoard()
+  setSecret()
+
+
+  
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  const value = questions.options[questions.selectedIndex].value;
 
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  // the value variable is a variable that stores the actual value of the question we've selected.
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
+  console.log(category, value)
 }
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
   const { category, value } = currentQuestion
-
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
 
+  /* here is an if statment to see what the user guessed on, and depending on that what the guessed the "filterCharacters" function will be 
+  invoke and filter the board (generateBoard) accordingly*/
+  let keep;
+  if (category === 'hair' || category === 'eyes') {
+    if (value === secret[category]) {
+      keep = true
+    } else {
+      keep = false
+    }
```

Comment:
A shorter way to write this would be to replace line 275-279 with:
`keep = value === secret[category];`
This will dynamically set `keep` to `true` or `false` depending on the condition